bug(windows): Error: Invalid or corrupt jarfile `node_modules\recheck-jar\package.json`

### 1. Summary

When I ran eslint-plugin-redos 4.6.0-beta.1 on Windows, I got the error:

```bat
G:\Kristinita>npx eslint KiraExampleReDoS.js
Error: Invalid or corrupt jarfile G:\Kristinita\node_modules\recheck-jar\package.json
```

### 2. MCVE

#### 2.1. Reproducibility

I reproduced the problem on remote Windows and didn’t reproduce it on remote Ubuntu.

#### 2.2. Remote CI builds

1. [**AppVeyor CI — Windows build**](https://ci.appveyor.com/project/Kristinita/sashatravis/builds/53027301)
1. [**Travis CI — Ubuntu build**](https://app.travis-ci.com/github/Kristinita/SashaTravis/builds/276749755)

#### 2.3. Installation

```bat
npm init

npm install --save-dev eslint eslint-plugin-redos@4.6.0-beta.1
```

#### 2.4. Files

##### 2.4.1. `eslint.config.mjs`

```js
import { defineConfig } from "eslint/config";
import redosPlugin from "eslint-plugin-redos";

export default defineConfig([redosPlugin.configs.flat.recommended]);

```

##### 2.4.2. `KiraExampleSafeRegEx.js`

```js
const kiraRegex = /^\d$/u;

```

##### 2.4.3. `KiraExampleReDoS.js`

```js
const kiraRegex = /^(a*)*$/u;

```

#### 2.5. Terminal commands

```bat
npx eslint KiraExampleSafeRegEx.js

npx eslint KiraExampleReDoS.js
```

#### 2.6. Behavior

##### 2.6.1. Expected — Travis CI

[**The line of the Travis CI build**](https://app.travis-ci.com/github/Kristinita/SashaTravis/builds/276749755#L211):

```bash
$ npx eslint KiraExampleSafeRegEx.js

The command "npx eslint KiraExampleSafeRegEx.js" exited with 0.

$ npx eslint KiraExampleReDoS.js

/home/travis/build/Kristinita/SashaTravis/KiraExampleReDoS.js

  1:19  error  Found a ReDoS vulnerable RegExp (exponential)  redos/no-vulnerable

✖ 1 problem (1 error, 0 warnings)

The command "npx eslint KiraExampleReDoS.js" exited with 1.
```

##### 2.6.2. Bug — AppVeyor CI

[**The line of the AppVeyor CI build**](https://ci.appveyor.com/project/Kristinita/sashatravis/builds/53027301#L43):

```bat
npx eslint KiraExampleSafeRegEx.js
Error: Invalid or corrupt jarfile C:\projects\sashatravis\node_modules\recheck-jar\package.json

npx eslint KiraExampleReDoS.js
Error: Invalid or corrupt jarfile C:\projects\sashatravis\node_modules\recheck-jar\package.json
C:\projects\sashatravis\KiraExampleReDoS.js
  1:19  error  Found a ReDoS vulnerable RegExp (exponential)  redos/no-vulnerable
✖ 1 problem (1 error, 0 warnings)
Command exited with code 1
```

#### 2.7. Environment

1. Operating system:

	1. Local — Windows 11 [Version 10.0.22621.3085]
	1. AppVeyor CI — Windows Server 2019
	1. Travis CI — Ubuntu 24.04.3 LTS Noble Numbat

1. Node.js 25.1.0
1. ESLint 9.39.1
1. eslint-plugin-redos 4.6.0-beta.1

Thanks.

--- a/packages/recheck/src/lib/exe.test.ts
+++ b/packages/recheck/src/lib/exe.test.ts
@@ -15,7 +15,7 @@ afterEach(() => {
 });
 
 test("jar", () => {
-  expect(exe.jar()).toMatch(/recheck-jar\/recheck\.jar$/);
+  expect(exe.jar()).toMatch(/recheck-jar[/\\]recheck\.jar$/);
 
   process.env["RECHECK_JAR"] = "x";
   expect(exe.jar()).toBe("x");
@@ -44,7 +44,7 @@ test("jar: invalid resolve (2)", () => {
 
 test("bin", () => {
   if (isSupported) {
-    expect(exe.bin()).toMatch(/recheck-\w+-\w+\/recheck(?:\.exe)?$/);
+    expect(exe.bin()).toMatch(/recheck-\w+-\w+[/\\]recheck(?:\.exe)?$/);
   }
 
   process.env["RECHECK_BIN"] = "x";

--- a/packages/recheck/src/lib/exe.ts
+++ b/packages/recheck/src/lib/exe.ts
@@ -13,7 +13,7 @@ export const jar: () => string | null = () => {
   try {
     const exe = __mock__require
       .resolve("recheck-jar/package.json")
-      .replace(/\/package\.json$/, "/recheck.jar");
+      .replace(/package\.json$/, "recheck.jar");
     return exe;
   } catch (err: any) {
     if (err && err.code == "MODULE_NOT_FOUND") {
@@ -60,9 +60,7 @@ export const bin: () => string | null = () => {
     /* c8 ignore next */
     const bin = isWin32 ? "recheck.exe" : "recheck";
     const pkg = `recheck-${os}-${cpu}/package.json`;
-    const exe = __mock__require
-      .resolve(pkg)
-      .replace(/\/package\.json$/, `/${bin}`);
+    const exe = __mock__require.resolve(pkg).replace(/package\.json$/, bin);
     return exe;
   } catch (err: any) {
     if (err && err.code == "MODULE_NOT_FOUND") {

